app/models/manageiq/providers/vmware/infra_manager/ovf_service.rb

```diff
@@ -0,0 +1,150 @@
+class ManageIQ::Providers::Vmware::InfraManager::OvfService < ServiceGeneric
+  delegate :template, :manager, :to => :service_template, :allow_nil => true
+
+  # A chance for taking options from automate script to override options from a service dialog
+  def preprocess(action, new_options = {})
+    return unless action == ResourceAction::PROVISION
+
+    if new_options.present?
+      _log.info("Override with new options:")
+      $log.log_hashes(new_options)
+    end
+
+    save_action_options(action, new_options)
+  end
+
+  def execute(action)
+    return unless action == ResourceAction::PROVISION
+
+    library_item_deploy_queue
+  end
+
+  def library_item_deploy_queue
+    task_options = {:action => "Deploying VMware Content Library Item", :userid => "system"}
+    queue_options = {
+      :class_name  => self.class.name,
+      :instance_id => id,
+      :method_name => "library_item_deploy",
+      :args        => {},
+      :role        => "ems_operations",
+      #:queue_name  => manager.queue_name_for_ems_operations,
+      :zone        => manager.my_zone
+    }
+
+    task_id = MiqTask.generic_action_with_callback(task_options, queue_options)
+    task = MiqTask.wait_for_taskid(task_id)
+    raise task.message unless task.status_ok?
+  end
+
+  def library_item_deploy(_options)
+    _log.info("OVF template provisioning with template ID: [#{template.id}] name:[#{template.name}] was initiated.")
+    opts = provision_options
+    _log.info("VMware Content Library OVF Tempalte provisioning with options:")
+    $log.log_hashes(opts)
+
+    @deploy_response = template.deploy(opts).to_hash
```

Comment:
Should this be setting the deploy_response in the `options` so that https://github.com/ManageIQ/manageiq-providers-vmware/pull/629/files#diff-8a81d9fdde17651511d24a06f3706760R89 will work?